packages/fastify-vite/config.js

```diff
@@ -136,12 +137,16 @@ function resolveRoot (root) {
   }
 }
 
-async function resolveViteConfig (root) {
+async function resolveViteConfig (root, dev) {
   for (const ext of ['js', 'mjs', 'ts', 'cjs']) {
     const configFile = join(root, `vite.config.${ext}`)
     if (exists(configFile)) {
       return [
-        await resolveConfig({ configFile }, 'build', 'production'),
+        await resolveConfig(
+          { configFile },
+          "build",
+          dev ? "development" : "production"
```

Comment:
avoid hardcoding build to production only (fixing HMR in React) etc....

Reply:
Ouch! Good catch!